go/libraries/doltcore/sqle/dfunctions/dolt_checkout.go

```diff
@@ -103,7 +104,21 @@ func DoDoltCheckout(ctx *sql.Context, args []string) (int, error) {
 	} else if isBranch {
 		err = checkoutBranch(ctx, dbName, name)
 		if errors.Is(err, doltdb.ErrWorkingSetNotFound) {
-			err = checkoutRemoteBranch(ctx, dbName, dbData, name)
+			// If there is a branch but there is no working set,
+			// somehow the local branch ref was created without a
+			// working set. This happened with old versions of dolt
+			// when running as a read replica, for example. Try to
+			// create a working set pointing at the existing branch
+			// HEAD and check out the branch again.
+			//
+			// TODO: This is all quite racey, but so is the
+			// handling in DoltDB, etc.
+			err = createWorkingSetForLocalBranch(ctx, dbData.Ddb, name)
```

Comment:
should we do in CLI?

Reply:
The CLI handles this differently, but this flow does work there. AFAICT nothing needed to change...